### PR TITLE
Feature/validator address

### PR DESCRIPTION
**Pull request by talekhinezh**
* Implement bech32 addressing for validators

### CODE REVIEW

**Review by siy — APPROVED** [review]
LGTM

**Review by Sajjon — CHANGES_REQUESTED** [review]
0 tests.... please add some none trivial tests for Bech32 and RRI and Validator address.

File: radixdlt-java-common/src/main/java/com/radixdlt/identifiers/Rri.java [hunk 1]
```diff
@@ -103,43 +102,14 @@ public static Rri fromBech32(String s) {
 		var d = Bech32.decode(s);
```
Comment (Sajjon) [inline radixdlt-java-common/src/main/java/com/radixdlt/identifiers/Rri.java:102]: `decode` can fail if not checksummed. Right now we just passthrough `org.bitcoinj.core.AddressFormatException`, I think it more produent to wrap this, and throw our own errors, saying that it was not checksummed (and all other errors)...

**Comment by Sajjon** [comment]
@talekhinezh needs more **_portable_** tests, I wrote some for you, because I needed them myself. Please improve as you see fit:

```java

public class Bech32Tests {

    private void doTestRRIBech32(String name, String expectedRRI, byte[] hash) {
        var rri = Rri.of(hash, name);


        Consumer<Rri> rriTester = rriToTest -> {
            assertEquals(expectedRRI, rriToTest.toString());
            assertEquals(Bytes.toHexString(hash), Bytes.toHexString(rriToTest.getHash()));
            assertEquals(name, rriToTest.getName());
        };

        rriTester.accept(rri);

        var rriFromString = Rri.fromBech32(expectedRRI);
        rriTester.accept(rriFromString);

        assertEquals(rri, rriFromString);
    }

    @Test
    public void testRRIBech32() {
        doTestRRIBech32("foobar", "foobar_rr1m6kmamckpjzlw", Bytes.fromHexString("deadbeef"));
        doTestRRIBech32("hello", "hello_rr1w3jhxar5v4ehguwx8gq", "testtest".getBytes(StandardCharsets.UTF_8));
    }

    @Test
    public void testGetDataFromBech32RRIString() {
        var str = "hello_rr1w3jhxar5v4ehguwx8gq";
        var rri = Rri.fromBech32(str);
        assertEquals("testtest", new String(rri.getHash()));
    }

    @Test(expected = AddressFormatException.class)
    public void testBech32RRIStringInvalidChecksum() {
        var res = Rri.fromBech32("hello_rr1w3jhxar5v4ehguwx8g3"); // "8g3" should have been "8gq";
        fail("Expected failure but got success.");
    }

    @Test
    public void testRRIFromSpecString() {
        var addressString = "9S8khLHZa6FsyGo634xQo9QwLgSHGpXHHW764D5mPYBcrnfZV6RT";
        var address = RadixAddress.from(addressString);
        var publicKey = address.getPublicKey();
        var rriResult = Rri.fromSpecString(	"/" + addressString + "/XRD");
        var rri = rriResult.toOptional().get();
        assertEquals("xrd_rr1qfumuen7l8wthtz45p3ftn58pvrs9xlumvkuu2xet8egzkcklqtesv2yq5l", rri.toString());
        var pubKeyHex = Bytes.toHexString(rri.getHash());
        assertEquals(Bytes.toHexString(publicKey.getCompressedBytes()), pubKeyHex);
    }

    @Test
    public void testValidatorBech32() {
        var keyPair = ECKeyPair.fromSeed(Bytes.fromHexString("deadbeef"));
        var publicKey = keyPair.getPublicKey();
        var validatorAddress = Address.ofValidator(publicKey);
        var expectedValidatorAddressString = "vb1qvx0emaq0tua6md7wu9c047mm5krrwnlfl8c7ws3jm2s9uf4vxcyvrwrazy";
        assertEquals(expectedValidatorAddressString, validatorAddress);

        var publicKeyFromValidatorAddress = Address.parseValidatorAddress(expectedValidatorAddressString);
        assertEquals(publicKey, publicKeyFromValidatorAddress);
    }
}

```

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/client/Address.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+		var convert = Bits.convertBits(bytes, 0, bytes.length, 8, 5, true);
+		return Bech32.encode("vb", convert);
+	}
+
+	public static ECPublicKey parseValidatorAddress(String v) {
+		var bech32Data = Bech32.decode(v);
+		if (!bech32Data.hrp.equals("vb")) {
+			throw new IllegalArgumentException();
+		}
+		var keyBytes = Bits.convertBits(bech32Data.data, 0, bech32Data.data.length, 5, 8, false);
```
Comment (Sajjon) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/client/Address.java:42]: I would not use `convertBits` inline like this, (5, 8, false) and (8, 5, true) are the tuple inputs we wanna use always when using Bech32, so I would rework this:

```java
	public static String ofValidator(ECPublicKey key) {
		var bytes = key.getCompressedBytes();
		var bech32Data = toBech32Data(bytes);
		return Bech32.encode("vb", bech32Data);
	}
	
	public static byte[] toBech32Data(byte[] bytes) {
		return Bits.convertBits(bytes, 0, bytes.length, 8, 5, true);
	}

	public static byte[] fromBech32Data(byte[] bytes) {
		return Bits.convertBits(bytes, 0, bytes.length, 5, 8, false);
	}

	public static ECPublicKey parseValidatorAddress(String v) {
		var bech32Data = Bech32.decode(v);
		if (!bech32Data.hrp.equals("vb")) {
			throw new IllegalArgumentException();
		}
		var keyBytes = fromBech32Data(bech32Data.data);
		try {
			return ECPublicKey.fromBytes(keyBytes);
		} catch (PublicKeyException e) {
			throw new IllegalStateException();
		}
	}
```

Comment (Sajjon) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/client/Address.java:42]: and maybe move `toBech32Data` and `fromBech32Data` to some better location...

**Pull request merged**